src/widgets/ActivityGroups/ui/ActivityCard/index.tsx

```diff
@@ -101,20 +102,45 @@ export const ActivityCard = ({ activityListItem }: Props) => {
 
   const restartActivity = () => {
     onStartActivity(true);
-    Mixpanel.track(MixEvents.ActivityRestarted, { [MixProperties.AppletId]: context.applet.id });
+
+    const analyticsPayload: MixpanelPayload = {
+      [MixpanelProps.AppletId]: context.applet.id,
+      [MixpanelProps.ActivityId]: activityListItem.activityId,
+    };
+
+    if (isFlow) {
+      analyticsPayload[MixpanelProps.ActivityFlowId] = activityListItem.flowId;
+    }
+
+    Mixpanel.track(MixpanelEvents.ActivityRestarted, analyticsPayload);
   };
+
   const resumeActivity = () => {
     onStartActivity(false);
-    Mixpanel.track(MixEvents.ActivityResumed, { [MixProperties.AppletId]: context.applet.id });
+
+    const analyticsPayload: MixpanelPayload = {
+      [MixpanelProps.AppletId]: context.applet.id,
+      [MixpanelProps.ActivityId]: activityListItem.activityId,
+    };
+
+    if (isFlow) {
+      analyticsPayload[MixpanelProps.ActivityFlowId] = activityListItem.flowId;
+    }
+
+    Mixpanel.track(MixpanelEvents.ActivityResumed, {
+      [MixpanelProps.AppletId]: context.applet.id,
+    });
```

Comment:
Unfortunately, this `restartActivity` function is also called when the user manually clicks the start activity button on the activity card (outside take now).

https://github.com/ChildMindInstitute/mindlogger-web-refactor/blob/af848c2bc1fcca901d09e14b333c745ae6c3ad7a/src/widgets/ActivityGroups/ui/ActivityCard/ActivityCardRestartResume.tsx#L81-L90

Which causes a resume event to get created right after the started one

<img width="480" alt="image" src="https://github.com/ChildMindInstitute/mindlogger-web-refactor/assets/14842108/4d7a3168-5430-4147-965c-de8bc5854390">

I think this is the existing behaviour, so it's totally fair if we want to ignore this. However, if we want to increase the fidelity of these events, we'll need to create a separate function that gets called for start activity

Reply:
Fixed this! Thanks for noticing!